charmcraft/extensions/registry.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2023 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Extension registry."""
+
+from typing import Dict, List
+
+from charmcraft import errors
+from charmcraft.extensions.extension import Extension
+
+_EXTENSIONS: Dict[str, Extension] = {}
+
+
+def get_extension_names() -> List[str]:
+    """Obtain a extension class given the name.
+
+    :param name: The extension name.
+    :return: The list of available extensions.
+    :raises ExtensionError: If the extension name is invalid.
+    """
+    return list(_EXTENSIONS.keys())
```

Comment:
I wonder if it wouldn't be more user-friendly to alphabetise this list?
```suggestion
    return sorted(_EXTENSIONS.keys())
```

Reply:
extensions are not going to be registered externally